### PR TITLE
Znapzend improvements

### PR DESCRIPTION
**Pull request by infinisil**
###### Motivation for this change
- Adds options `features.compressed` to use `-Lce` for sending and `features.recvu` to allow destination filesystems to be unmounted. See `man znapzend` for info on these features.
- Fixes the service timing out when starting when a destination host is offline.

Ping @lheckemann 

###### Things done

- [x] Made sure both commits have the intended changes in my local system

### CODE REVIEW

**Comment by lheckemann** [comment]
Could we also enable them by default (with a release notes entry :wink:)?

**Comment by lheckemann** [comment]
(and thanks very much for implementing this!)

**Comment by Infinisil** [comment]
@lheckemann I thought about it, but enabling them could break for people who have e.g. `embedded_data` enabled on the sending side, but disabled on the receiving one, same for other features. I think by default these are off in znapzend because of this too.

**Comment by lheckemann** [comment]
Hence the release notes entry ;)

But let's start with merging this and consider changing it separately.

File: nixos/modules/services/backup/znapzend.nix [hunk 1]
```diff
@@ -381,12 +399,22 @@ in
         '';
 
         serviceConfig = {
+          # znapzendzetup --import apparently tries to connect to the backup
+          # host 3 times with a timeout of 30 seconds, leading to a startup
+          # delay of >90s when the host is down, which is just above the default
+          # service timeout of 90 seconds. Increase the timeout so it doesn't
+          # make the service fail in that case.
+          TimeoutStartSec = 180;
+          # Needs to have write access to ZFS
```
Comment (lheckemann) [inline nixos/modules/services/backup/znapzend.nix]: That doesn't necessarily require root (https://github.com/oetiker/znapzend/blob/master/README.md#running-by-an-unprivileged-user)

**Comment by lheckemann** [comment]
In retrospect, this should probably be a per-destination property since different destination pools may support different features.

**Comment by Infinisil** [comment]
@lheckemann Unfortunately znapzend doesn't support this :/

**Pull request merged**